barril/basic/format_float/__init__.py

```diff
@@ -0,0 +1,129 @@
+PLUS_INFINITY_STR = "+INF"
+
+MINUS_INFINITY = float('-inf')
+MINUS_INFINITY_STR = "-INF"
+
+NAN = float('nan')
+NAN_STR = "-1.#IND"
+
+
+#===================================================================================================
+# FormatFloat
+#===================================================================================================
+def FormatFloat(format, value, grouping=False, use_locale=True):
+    '''
+    Formats the value given according to the current LC_NUMERIC setting. The format follows the
+    conventions of the % operator. For floating point values, the decimal point is modified if
+    appropriate. If grouping is True, the thousands separator set in locale settings is also taken
+    into account.
+
+    @note:
+        This function is used to format float values. There are known issues like convert unit 
+        from scalar where value is "0.0", it can get representation "-0.0"
+
+    :param unicode format:
+        The format used to format the value.
+
+    :param float value:
+        The value to be formated.
+
+    :param bool grouping:
+        True if the thousands separator must be used on formating the number. The locale settings
+        must be explicitly set in order to make sure the thousands separator will be applied. The
+        default value for thousands separator is "", so that the value formated won't have thousands
+        separator.
+
+    :param bool use_locale:
+        Use locale.format or unicode % operator (locale-independent output).
+
+    :rtype: unicode
+    :returns:
+        The formated value.
+    '''
+    # Handling INFINITY
+    # - locale.format tries to round the infinity value:
+    #   ( ".3g", 9e999 ) ==> +1.#J
+    import locale
+
+    if value == PLUS_INFINITY:
+        return PLUS_INFINITY_STR
+    elif value == MINUS_INFINITY:
+        return MINUS_INFINITY_STR
+    elif value != value:  # Not a Number
+        # numpy.isnan(x) is more readable, but it doesn't work here because FractionValue-s seem to
+        # pass numpy arrays to FormatFloat, and numpy.isnan doesn't accept them.
+        return NAN_STR
+
+    # This removes the minus sign from the string representation of the float
+    if value == 0.0:
+        value = 0.0
+
+    if use_locale:
+        try:
+            result = locale.format(format, value, grouping)
+        except TypeError:
+            # Python has a limitation to convert large float numbers to integer format.
+            # To avoid this the values will be forced to int when requesting for the integer format.
+            result = locale.format(format, int(value), grouping)
+    else:
+        result = format % value
+
+    return result
+
+
+#===================================================================================================
+# FloatFromString
+#===================================================================================================
+def FloatFromString(str_value, use_locale=True):
```

Comment:
Did we decide if we are going to rename all methods to PEP-8?

Reply:
No!